stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -138,15 +146,34 @@ internal class PaymentOptionsActivity : BaseSheetActivity<PaymentOptionResult>()
                 }
             }
         }
+    }
 
-        viewModel.userSelection.observe(this) { paymentSelection ->
-            if (paymentSelection != null) {
-                onActionCompleted(paymentSelection)
+    @VisibleForTesting
+    private val viewStateObserver: (ViewState.PaymentOptions?) -> Unit
+        get() = { viewState ->
+            val addButton = viewBinding.addButton
+            when (viewState) {
+                is ViewState.PaymentOptions.Ready -> addButton.updateState(
+                    state = PrimaryButton.State.Ready(addButtonLabel)
+                )
+                is ViewState.PaymentOptions.Confirming -> addButton.updateState(
+                    state = PrimaryButton.State.Confirming
+                )
+                is ViewState.PaymentOptions.Completed -> addButton.updateState(
+                    state = PrimaryButton.State.Completed,
+                    completeCallback = {
+                        onActionCompleted(viewState.result)
+                    }
+                )
+                is ViewState.PaymentOptions.Finished -> onActionCompleted(viewState.result)
```

Comment:
nit: you could simplify this to

```
@VisibleForTesting
private val viewStateObserver = { viewState: ViewState.PaymentOptions? ->
   ...
}
```

Reply:
Actually that makes more sense to me, currently this is creating a new instance of the lambda every time `viewStateObserver` is accessed.

Reply:
that makes sense